recordedit/recordEdit.app.js

```diff
@@ -335,13 +335,28 @@
                                         default:
                                             // the structure for asset type columns is an object with a 'url' property
                                             if (column.isAsset && colModel.inputType !== "disabled") {
+                                                var metadata = column.getMetadata(tuple.data);
                                                 value = {
                                                     url: values[i] || "",
-                                                    filename: tuple.data[column._filenameColumn.name]
+                                                    filename: metadata.filename,
+                                                    filesize: metadata.byteCount
                                                 };
                                             } else {
                                                 value = values[i];
                                             }
+
+                                            // if in copy mode and copying an asset column with metadata available, attach that to the submission model
+                                            if (column.isAsset && context.mode == context.modes.COPY) {
+                                                // may not have been set or fetched above because of disabled case
+                                                // we still want to copy the metadata
+                                                var metadata = column.getMetadata(tuple.data);
+
+                                                // I don't think this should be done brute force like this
+                                                if (metadata.filename) recordEditModel.submissionRows[j][column.filenameColumn.name] = metadata.filename;
```

Comment:
I could add a new API to the `getMetadata` because the `filename` that I'm returning is not necessarily the filename but rather the name that we can use to refer to the file. I can change its name to something else.

Also `column.filenameColum` and the rest might be undefined.

Reply:
The previous comment is not relevant anymore. After discussing about this last week, I separated the displayed value from the filename. So 
- `filename` will return the filename that is stored in database.
- `caption` will return the value that should be displayed for the asset

So as long as we make sure that we're storing the `metadata.filename` and showing `metadata.caption`, it should be fine.